src/main/java/pl/betoncraft/betonquest/Instruction.java

```diff
@@ -466,12 +466,12 @@ public double getDouble(final String string, final double def) throws Instructio
     }
 
     public ArrayList<Integer> getAllNumbers() {
-        final Pattern p = Pattern.compile("-?\\d+");
+        final Pattern p = Pattern.compile("\\b[\\d]+\\b");
         final Matcher m = p.matcher(instruction);
 
         final ArrayList<Integer> result = new ArrayList<>();
         while (m.find()) {
-            result.add(Integer.parseInt(m.group()));
+            result.add(Integer.parseInt(m.group().replaceAll(" ","")));
```

Comment:
As spaces will never be matched by any of the old or new patterns this replaceAll execution is useless.

Reply:
Nice catch!